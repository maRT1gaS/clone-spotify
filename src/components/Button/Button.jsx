import React, { useState } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './Button.module.css';

export const Button = ({ children, type, onClick }) => {
  const [onTab, setOnTab] = useState(false);

  const setOutline = (key) => {
    if (key.code === 'Tab') {
      setOnTab(true);
    }
  };
  return (
    <button
      type={type === 'button' ? 'button' : 'submit'}
      className={cn(styles.button, {
        [styles.onTab]: onTab,
      })}
      onBlur={() => setOnTab(false)}
      onKeyUp={setOutline}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.oneOf(['submit', 'button']),
  onClick: PropTypes.func,
};

Button.defaultProps = {
  type: 'button',
  onClick: null,
};
