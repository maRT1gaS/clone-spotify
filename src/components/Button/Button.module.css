@import '../../styles/variables.css';

.button {
  padding: 0.5rem 5rem;

  cursor: pointer;

  transition: 0.4s ease;

  border-radius: 10rem;
  background-color: var( --white );
}

.button:hover {
  transform: scale( 1.1 );

  background-color: var( --green );

  font-weight: 700;
}

.isFocus {
  transform: scale( 1.1 );

  background-color: var( --green );

  font-weight: 700;
}
