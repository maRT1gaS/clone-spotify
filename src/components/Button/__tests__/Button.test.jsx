import { shallow } from 'enzyme';
import React from 'react';

import { Button } from '../Button';

describe('Button', () => {
  it('should render correctly', () => {
    const component = shallow(<Button>Войти</Button>);
    expect(component).toMatchSnapshot();
  });
  // it('button click should clicked', () => {
  //   const clickFn = jest.fn();
  //   const component = shallow(<Button onClick={clickFn}>Войти</Button>);

  //   component.find()
  // });
});
