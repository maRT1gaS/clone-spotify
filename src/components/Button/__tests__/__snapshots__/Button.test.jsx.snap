// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="button"
  onBlur={[Function]}
  onClick={null}
  onKeyUp={[Function]}
  type="button"
>
  Войти
</button>
`;
